pkg/executables/kind.go

```diff
@@ -165,7 +169,7 @@ func (k *Kind) WithEnv(env map[string]string) bootstrapper.BootstrapClusterClien
 	}
 }
 
-func (k *Kind) WithRegistryMirror(endpoint string, caCertFile string) bootstrapper.BootstrapClusterClientOption {
+func (k *Kind) WithRegistryMirror(endpoint string, caCertFile string, env map[string]string) bootstrapper.BootstrapClusterClientOption {
```

Comment:
Can we avoid using `map[string]string` to pass credentials and toggle registry auth configuration? It makes for a hard to understand API.

We might want to shift to a `RegistryMirror` configuration object and describe what's optional etc making expectations very clear.

E.g.

```go
type RegistryMirrorAuth struct {
  Username string
  Password string
}

type RegistryMirrorConfig struct {
  Endpoint   string
  CACertPath string
  *RegistryMirrorAuth
}

func (k *Kind) WithRegistryMirror(RegistryMirrorConfig) bootstrapper.BootstrapClusterClientOption {
```